libmamba/src/core/fetch.cpp

```diff
@@ -260,18 +260,34 @@ namespace mamba
             curl_easy_setopt(handle, CURLOPT_SSL_OPTIONS, CURLSSLOPT_NO_REVOKE);
         }
 
+        const char* proxy = proxy_match(url);
```

Comment:
I guess it would be preferable to use `std::string` whenever possible.

Reply:
I just copied that from #161, as I thought there might be a reason for using `char*`. I can try to change it tough.
The problem is that `std::string` isn't allowed to be `nullptr` and using empty string instead feels a bit hacky. I might be able to use `std::optional` though.